extensions/amp-analytics/0.1/visibility-impl.js

```diff
@@ -149,37 +150,38 @@ export function isVisibilitySpecValid(config) {
  * id, the element with that id is returned. If the selector is a tag name, an
  * ancestor of the analytics element with that tag name is returned.
  *
+ * @param {!../../../src/service/ampdoc-impl.AmpDoc} ampdoc.
  * @param {string} selector The selector for the element to track.
- * @param {!Element} el Element whose ancestors to search.
- * @param {!String} selectionMethod The method to use to find the element..
+ * @param {!Element} analyticsEl Element whose ancestors to search.
+ * @param {!String} selectionMethod The method to use to find the element.
  * @return {?Element} Element corresponding to the selector if found.
  */
-export function getElement(ampdoc, selector, el, selectionMethod) {
-  if (!el) {
+export function getElement(ampdoc, selector, analyticsEl, selectionMethod) {
+  if (!analyticsEl) {
     return null;
   }
 
+  let foundEl;
+  const friendlyFrame = getParentWindowFrameElement(analyticsEl, ampdoc.win);
   // Special case for root selector.
-  if (selector == ':host' || selector == ':root') {
-    const elWin = ampdoc.win;
-    const parentEl = elWin.frameElement && elWin.frameElement.parentElement;
-    if (parentEl) {
-      return closestBySelector(parentEl, '.-amp-element');
-    }
+  if (friendlyFrame && (selector == ':host' || selector == ':root')) {
+    foundEl = closestBySelector(friendlyFrame, '.-amp-element');
```

Comment:
I believe we should do this:
```
if (selector == :host || selector == :root) {
  foundEl = friendlyFrame ? closestBySelector(...) : null;
}
```

Otherwise the code in `else` below does not actually produce the right result. We have a follow up issue for @avimehta to address this, but to start with it should not produce a wrong result.